src/rest.rs

```diff
@@ -1128,6 +1128,26 @@ fn handle_request(
 
             json_response(prepare_txs(txs, query, config), TTL_SHORT)
         }
+        (&Method::POST, Some(&"mempool"), Some(&"txs"), None, None, None) => {
+            let txid_strings: Vec<String> =
+                serde_json::from_slice(&body).map_err(|err| HttpError::from(err.to_string()))?;
+
+            match txid_strings
+                .into_iter()
+                .map(|txid| Txid::from_hex(&txid))
+                .collect::<Result<Vec<Txid>, HashHexError>>()
+            {
+                Ok(txids) => {
+                    let txs: Vec<(Transaction, Option<BlockId>)> = txids
+                        .iter()
+                        .filter_map(|txid| query.mempool().lookup_txn(txid).map(|tx| (tx, None)))
+                        .collect();
```

Comment:
`query.mempool()` takes a RwLock every time it's called, and since lookup_txn only uses &self, you can pass a reference of the RwLockReadGuard into the filter_map closure without issue.

```suggestion
                    let txs: Vec<(Transaction, Option<BlockId>)> = {
                        let mempool = query.mempool();
                        txids
                            .iter()
                            .filter_map(|txid| mempool.lookup_txn(txid).map(|tx| (tx, None)))
                            .collect()
                    };
```